File: Android/Java/UD3/UD03EJ012J_Receive/app/src/main/java/com/pmdm/ud03ej012j_receive/MainActivity.java
package com.pmdm.ud03ej012j_receive;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Bundle extras = getIntent().getExtras();
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TextView vistaTexto = findViewById(R.id.TextViewMayusculas);
        vistaTexto.setText(extras.getString(Intent.EXTRA_TEXT).toUpperCase());
    }

    public void salir(View view){
        finish();
    }
}

File: Android/Kotlin/UD3/UD03EJ012K_Receive/app/src/main/java/com/pmdm/ud03ej012k_receive/MainActivity.kt
package com.pmdm.ud03ej012k_receive

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        var extras: Bundle? = getIntent().getExtras()
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var vistaTexto: TextView = findViewById(R.id.TextViewMayusculas)

        if(extras != null) {
            var texto: String? = extras.getString(Intent.EXTRA_TEXT)

            if (texto != null) {
                vistaTexto.setText((texto.uppercase()))
            }
        }
    }

    fun salir(view: View){
        finish()
    }
}